lib/membrane/video_compositor/frame_compositor.ex

```diff
@@ -31,19 +39,29 @@ defmodule Membrane.VideoCompositor.FrameCompositor do
               id :: id_t(),
               input_caps :: RawVideo.t(),
               position :: {x :: non_neg_integer(), y :: non_neg_integer()},
-              z :: float()
+              z :: float(),
+              scale :: float()
             ) :: {:ok, internal_state_t} | {:error, error_t()}
 
   @doc """
-  Video of the given `id` should be registered, removal of nonexistent video may cause undefined behaviour.
+  Video of the given `id` should be registered, removal of nonexistent video may panic the compositor.
   """
   @callback remove_video(
               internal_state :: internal_state_t,
               id :: id_t()
             ) :: {:ok, internal_state_t} | {:error, error_t()}
 
   @doc """
-  Video of the given `id` should be registered, using `id` of nonexistent video may cause undefined behaviour.
+  Send an end of stream to a video with the given `id`.
+
+  This causes the video to be deleted after it's enqueued frames are used up.
+  """
+  @callback send_end_of_stream(internal_state_t(), id_t()) ::
+              {:ok, internal_state_t()} | {:error, error_t()}
+
+  @doc """
+  Video of the given `id` should be registered, using `id` of nonexistent video may panic the compositor.
```

Comment:
Compositor should return `{:error, error}` in case of setting the position of a non-existing video.

Reply:
this is outside of the scope of this (already huge) PR. I will not fix it here.